zh_cn/news/_posts/2019-05-30-ruby-2-7-0-preview1-released.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: news_post
+title: "Ruby 2.7.0-preview1 已发布"
+author: "naruse"
+translator: "Delton Ding"
+date: 2019-05-30 00:00:00 +0000
+lang: zh_cn
+---
+
+我们很高兴宣布 Ruby 2.7.0-preview1 已发布。
+
+此版本的发布是为了十二月发布的正式版收集反馈意见。此版本引入了大量新特性和性能提升，其中最值得注意的是：
+
+* 紧凑 GC（Compaction GC）
+* 模式匹配
+* REPL 改进
+
+## 紧凑 GC（Compaction GC）
+
+此版本引入了 Compaction GC，可以对内存空间碎片进行整理。
+
+某些多线程 Ruby 程序会导致内存碎片化，进而导致内存占用率提高和速度降低。
+
+我们引入了 `GC.compact` 方法来压缩堆。此函数能压缩堆中的存活对象，以更少地占用内存分页。并且堆可能会变得对写入时复制（CoW）更友好。
+[#15626](https://bugs.ruby-lang.org/issues/15626)
+
+## 模式匹配 [实验性]
+
+在函数式编程中非常常用的模式匹配功能，作为实验性功能被加入了。[#14912](https://bugs.ruby-lang.org/issues/14912) 它可以遍历一个对象，并在其满足某一模式时进行赋值。
+
+```ruby
+case JSON.parse('{...}', symbolize_names: true)
+in {name: "Alice", children: [{name: "Bob", age: age}]}
+  p age
+  ...
+end
+```
+
+关于更多信息，请查阅 [Pattern matching - New feature in Ruby 2.7](https://speakerdeck.com/k_tsj/pattern-matching-new-feature-in-ruby-2-dot-7)。
+
+## REPL 改进
+
+`irb`，集成的交互环境 (REPL; Read-Eval-Print-Loop)，现已支持多行编辑，由 `reline`、`readline` 实现了纯 Ruby 的支持。它还提供 rdoc 集成。在 irb 中您可以为指定的类、模块或方法的显示引用。[#14683](https://bugs.ruby-lang.org/issues/14683)、[#14787](https://bugs.ruby-lang.org/issues/14787)、[#14918](https://bugs.ruby-lang.org/issues/14918) 此外，`binding.irb` 中显示的代码和黑心类的检查结果现在已经可以彩色呈现。
+
+<video autoplay="autoplay" controls="controls" muted="muted" width="576" height="259">
+  <source src="https://cache.ruby-lang.org/pub/media/irb_improved_with_key_take2.mp4" type="video/mp4">
+</video>
+
+## 其它值得关注的新特性
+
+* 方法引用运算符，<code>.:</code>，作为实验性功能加入了。[#12125]( https://bugs.ruby-lang.org/issues/12125)、[#13581]( https://bugs.ruby-lang.org/issues/13581)
+
+* 实验性地加入了把编号参数作为默认的块参数的特性。[#4475](https://bugs.ruby-lang.org/issues/4475)
+
+* 无头范围实验性地加入了。它可能尽管没有无限范围那么有用，但它对开发 DSL 是非常有用的。[#14799](https://bugs.ruby-lang.org/issues/14799)
+
+      ary[..3]  # 与 ary[0..3]
+      rel.where(sales: ..100) 等价
+
+* 新增了 `Enumerable#tally`，它会计算每个元素出现的次数。
+
+      ["a", "b", "c", "b"].tally
+      #=> {"a"=>1, "b"=>2, "c"=>1}
+
+## 性能改进
+
+* JIT [实验性质]
+
+  * 当优化假设不成功时，JIT 后的代码可能会被重新编译到优化程度较低的代码。
```

Comment:
extra leading spaces?

Reply:
I'm sorry that I don't get it. Which space is the extra leading space?

Reply:
Sorry my mistake, looks good to me.